src/NUnitFramework/framework/Attributes/TestCaseSourceAttribute.cs

```diff
@@ -243,7 +243,7 @@ private IEnumerable<ITestCaseData> GetTestCasesFor(IMethodInfo method)
             return data;
         }
 
-        private IEnumerable GetTestCaseSource(IMethodInfo method)
+        protected virtual IEnumerable GetTestCaseSource(IMethodInfo method)
```

Comment:
One thing that bothers me with the approach of overriding the `TestCaseSourceAttribute` is that you are not calling to this base from the derived classes. Because of that, I don't feel that this is quite the right implementation. I think that if you want to take this approach, then you should move everything in `TestCaseSourceAttribute` into a new abstract base class `BaseTestCaseSourceAttribute` making the `GetTestCaseSource` method abstract. Then each of the source attributes would derive from that.